### PR TITLE
ENH support custom namespaces

### PR DESCRIPTION
**Pull request by beckermr**
This PR adds support for custom namespaces. 

Before

```
$ depfinder metadetect
{'builtin': ['__future__',
             'contextlib',
             'copy',
             'logging',
             'os',
             'pdb',
             'sys',
             'time',
             'warnings'],
 'questionable': ['astropy',
                  'biggles',
                  'descwl_coadd',
                  'images',
                  'lsst',
                  'matplotlib',
                  'scarlet',
                  'sxdes'],
 'relative': ['_version',
              'configs',
              'defaults',
              'detect',
              'fitting',
              'interpolate',
              'masking',
              'measure',
              'metacal_exposures',
              'mfrac',
              'procflags',
              'sim',
              'skysub',
              'util'],
 'required': ['descwl_coadd',
              'descwl_shear_sims',
              'esutil',
              'galsim',
              'lsst',
              'matplotlib',
              'meds',
              'ngmix',
              'numba',
              'numpy',
              'pytest',
              'scipy',
              'tqdm']}
```

and after

```
$ depfinder --custom-namespaces=lsst metadetect
{'builtin': ['__future__',
             'contextlib',
             'copy',
             'logging',
             'os',
             'pdb',
             'sys',
             'time',
             'warnings'],
 'questionable': ['astropy',
                  'biggles',
                  'descwl_coadd',
                  'images',
                  'lsst.afw',
                  'lsst.geom',
                  'lsst.meas',
                  'lsst.pex',
                  'lsst.pipe',
                  'matplotlib',
                  'scarlet',
                  'sxdes'],
 'relative': ['_version',
              'configs',
              'defaults',
              'detect',
              'fitting',
              'interpolate',
              'masking',
              'measure',
              'metacal_exposures',
              'mfrac',
              'procflags',
              'sim',
              'skysub',
              'util'],
 'required': ['descwl_coadd',
              'descwl_shear_sims',
              'esutil',
              'galsim',
              'lsst.afw',
              'lsst.geom',
              'lsst.meas',
              'lsst.pex',
              'matplotlib',
              'meds',
              'ngmix',
              'numba',
              'numpy',
              'pytest',
              'scipy',
              'tqdm']}
```

### CODE REVIEW

**Comment by beckermr** [comment]
@CJ-Wright Care to review and merge?

**Comment by CJ-Wright** [comment]
The tests seem unhappy. I'm not a fan of the global but I can merge with it if you want to keep it.

**Comment by beckermr** [comment]
I'll take a look. I matched the global to the other namespace handling in the code. I can pass a function argument all the way through as well.

**Comment by CJ-Wright** [comment]
Ok, I leave that to your judgement. Happy to merge once the tests pass. Will you want a release once merged?

I am making this comment solely in my personal capacity and am not conveying any rights to any intellectual property of any third parties.

**Comment by beckermr** [comment]
Yeah a release would be good. I'll bump you when I'm ready. Thank you!

**Comment by beckermr** [comment]
ok @CJ-Wright 

I removed the global and got the tests to pass locally! Can you enabled the workflow again and merge when you are happy?

File: depfinder/utils.py [hunk 1]
```diff
@@ -60,4 +60,8 @@
         Loader=yaml_loader,
     )
 
-namespace_packages = {pkg['import_name'] for pkg in mapping_list if '.' in pkg['import_name']}
\ No newline at end of file
+namespace_packages = {pkg['import_name'] for pkg in mapping_list if '.' in pkg['import_name']}
+
+# this global holds custom namespaces for namespace packages (e.g., a set of
+# packages like foo.bar and foo.baz where foo itself is not a package)
+custom_namespaces = []
```
Comment (beckermr) [inline depfinder/utils.py:67]:
```suggestion
```

**Comment by beckermr** [comment]
Ready to merge @CJ-Wright? The tests pass but the coverage diff went down a bit.

**Comment by CJ-Wright** [comment]
Thank you very much!


I am making this comment solely in my personal capacity and am not conveying any rights to any intellectual property of any third parties.

**Pull request merged**